certbot/plugins/dns_cloudflare.py

```diff
@@ -0,0 +1,323 @@
+"""DNS Authenticator for Cloudflare."""
+import logging
+
+import zope.interface
+
+from time import sleep
+
+from acme import challenges
+
+from certbot import errors
+from certbot import interfaces
+
+from certbot.display import util as display_util
+
+from certbot.plugins import common
+
+import CloudFlare
+
+logger = logging.getLogger(__name__)
+
+
+@zope.interface.implementer(interfaces.IAuthenticator)
+@zope.interface.provider(interfaces.IPluginFactory)
+class Authenticator(common.Plugin):
+    """DNS  Authenticator for Cloudflare
+
+    This Authenticator uses the Cloudflare API to fulfill a dns-01 challenge.
+    """
+
+    description = 'Obtain certs using a DNS TXT record (if you are using Cloudflare for DNS).'
+
+    propagation_seconds = 10
```

Comment:
10 seconds was consistently enough for my testing, but we can easily increase this if we see it taking longer.